third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb

```diff
@@ -0,0 +1,183 @@
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"on_delete": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"cancel", "drain"}, false),
+				Optional:     true,
+				Default:      "drain",
+			},
+
+			"labels": {
+				Type:             schema.TypeMap,
+				Optional:         true,
+				DiffSuppressFunc: resourceDataflowJobLabelDiffSuppress,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				ForceNew: true,
+				Optional: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				// ForceNew applies to both stream and batch jobs
+				ForceNew: true,
+			},
+
+			"job_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// resourceDataflowFlexTemplateJobCreate creates a Flex Template Job from TF code.
+func resourceDataflowFlexTemplateJobCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	request := dataflow.LaunchFlexTemplateRequest{
+		LaunchParameter: &dataflow.LaunchFlexTemplateParameter{
+			ContainerSpecGcsPath: d.Get("container_spec_gcs_path").(string),
+			JobName:              d.Get("name").(string),
+			Parameters:           expandStringMap(d, "parameters"),
+		},
+	}
+	response, err := config.clientDataflow.Projects.Locations.FlexTemplates.Launch(project, region, &request).Do()
+	if err != nil {
+		return err
+	}
+	job := response.Job
+	d.SetId(job.Id)
+	d.Set("job_id", job.Id)
+
+	return resourceDataflowFlexTemplateJobRead(d, meta)
+}
+
+// resourceDataflowFlexTemplateJobRead reads a Flex Template Job resource.
+func resourceDataflowFlexTemplateJobRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	jobId := d.Id()
+
+	job, err := resourceDataflowJobGetJob(config, project, region, jobId)
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Dataflow job %s", jobId))
+	}
+
+	d.Set("state", job.CurrentState)
+	d.Set("name", job.Name)
+	d.Set("project", project)
+	d.Set("labels", job.Labels)
+
+	if _, ok := dataflowTerminalStatesMap[job.CurrentState]; ok {
+		log.Printf("[DEBUG] Removing resource '%s' because it is in state %s.\n", job.Name, job.CurrentState)
+		d.SetId("")
+		return nil
+	}
+	d.SetId(job.Id)
+	d.Set("job_id", job.Id)
+
+	return nil
+}
+
+// resourceDataflowFlexTemplateJobUpdateByReplacement updates a Flex Template job by replacement.
+func resourceDataflowFlexTemplateJobUpdateByReplacement(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Is there documentation available on what pipeline update will look like? Changing the update method under-the-hood could count as a breaking change, so it may be better to release without update and add that once it's available (which would just be an enhancement).

Otherwise, issues > TODOs 🙂

Reply:
+1 to releasing without an update. Before adding Terraform support for update-by-replacement, the standard Dataflow job implementation had no update method and used `ForceNew` on every field.